service/raft/raft_group0_client.hh

```diff
@@ -190,4 +191,54 @@ public:
     void set_query_result(utils::UUID query_id, service::broadcast_tables::query_result qr);
 };
 
+using mutations_generator = coroutine::experimental::generator<mutation>;
+
+// mutations_collector is used to gather mutations which are side effects
+// of functions execution. They need to be announced under single guard
+// for atomicity. As functions which produce mutations may embed each other
+// we need to decouple announcing step to a common external place here.
+// It also supports generator callbacks to avoid holding too many mutations
+// in memory.
+//
+// Single mutations_collector object represents a single transaction.
+// If size or number of mutations is too big for raft too handle it will be
+// rejected.
+class mutations_collector {
```

Comment:
Smells like it could be named group0_transaction.

Reply:
in theory it's not bound to group0 specifically

Reply:
It has group0 all over it.

When we develop multi-table transactions over strongly consistent tables managed by raft, we can decide whether to reuse this class, or create a new one.

Reply:
group0_batch perhaps or group0_mutation_batch

Reply:
I can rename. But where exactly it has group0 all over it?

Reply:
the guard as a member?